dash/backend/src/modules/falco/service/falco.service.ts

```diff
@@ -78,13 +77,28 @@ export class FalcoService {
         return newResultSet;
     }
 
-    async getFalcoCsv( clusterId: number): Promise<FalcoCsvDto> {
-        const queryResponse = await this.falcoDao.getFalcoLogsForExport(clusterId);
+    async getFalcoCsv( clusterId: number,
+                       limit = 20,
+                       page = 0,
+                       priorities?: string [],
+                       orderBy?: string,
+                       startDate?: string,
+                       endDate?: string,
+                       namespace?: string,
+                       pod?: string,
+                       image?: string,
+                       signature?: string,
+                       ): Promise<FalcoCsvDto> {
+
+        // retrieve filtered falco logs and use the query results to build the csv
+        const queryResp = await this.falcoDao.getFalcoCsvLogs(clusterId, priorities, orderBy, startDate, endDate, namespace, pod, image,);
         const result = [this.csvService.buildLine(['Date', 'Namespace', 'Pod',
             'Image', 'Priority', 'Message'])];
 
-        for (let i = 0; i < queryResponse.logCount; i++) {
-            const falcoCol = queryResponse.fullList[i];
+        // limit to 1000 or less logs
+        const numOfLogs = queryResp.csvLogList.length <= 1000 ? queryResp.csvLogList.length : 1000;
```

Comment:
Now that we are limiting the resultset size properly in the DAO, you can simplify this code to just a few lines: 

    const result = queryResp.csvLogList.map(falcoCol => this.csvService.buildLine([
                String(falcoCol.calendarDate),
                String(falcoCol.namespace),
                String(falcoCol.container),
                String(falcoCol.image),
                String(falcoCol.level),
                String(falcoCol.message),
            ]));